iceoryx_examples/iceperf/iceoryx.cpp

```diff
@@ -68,24 +69,29 @@ void Iceoryx::shutdown() noexcept
 
 void Iceoryx::sendPerfTopic(uint32_t payloadSizeInBytes, bool runFlag) noexcept
 {
-    auto sendSample = static_cast<PerfTopic*>(m_publisher.allocateChunk(payloadSizeInBytes, true));
-    sendSample->payloadSize = payloadSizeInBytes;
-    sendSample->run = runFlag;
-    sendSample->subPackets = 1;
-
-    m_publisher.sendChunk(sendSample);
+    m_publisher.loan(payloadSizeInBytes).and_then([&](iox::popo::Sample<void>& sample) {
+        auto sendSample = static_cast<PerfTopic*>(sample.get());
+        sendSample->payloadSize = payloadSizeInBytes;
+        sendSample->run = runFlag;
+        sendSample->subPackets = 1;
+        sample.publish();
+    });
 }
 
 PerfTopic Iceoryx::receivePerfTopic() noexcept
 {
     const void* receivedChunk;
-    while (!m_subscriber.getChunk(&receivedChunk))
+    PerfTopic receivedSample;
+    while (!m_subscriber.take().and_then([&](iox::cxx::optional<iox::popo::Sample<const void>>& maybeSample) {
+        if (maybeSample.has_value())
+        {
+            receivedSample = *(static_cast<const PerfTopic*>(maybeSample.value().get()));
```

Comment:
don't we need to release the ownership from `maybeSample` after assigning the sample to `receivedSample`?

Reply:
With the latest change, you will directly get the sample here and not an optional.
i.e. No more maybe.